### DISCUSSION TITLE
Remeshing to obtain exactly N vertices

### DISCUSSION THREAD

**Discussion by jcxz100h** [libigl/libigl#2264] [Q&A] [answered]
Dear all, first let me thank you for the library! I have moved from CGAL and I find it really refreshing.

I have a question regarding remeshing: my objective is to remesh obtaining **exactly** N vertices. Of course, keeping the shape as close as possible to the original (or in fortunate cases, exactly).

I have seen `upsample` and `decimate`, but they won't fit the bill, in particular `decimate` tends to modify the shape of the mesh. For instance, I have a standard 8-vertex cube, and I'd like to obtain N points, for example 1024, an upsampling with a subsequent decimation tends in the last phase to remove vertices on the edge of the cube.

How would you suggest to achieve this? Is there for instance any function that can subdivide a single triangle without doing by-hand calculations?

Thank you!

**Comment by alecjacobson** [comment, marked as answer]
If you're always _increasing_ the number of vertices then maybe you could try `upsample` followed by `simplify_polyhedron`. Might end up with some really skinny triangles though.

Is it important to have triangles? Or just points? blue_noise would be a good way to just get points. 

In a pinch sometimes I've tried running blue_noise with n = target_num - #V and then inserting those points into the mesh. This also results in awfully skinny triangles.

**Reply by jcxz100h** [reply]
@alecjacobson Thank you very much for your answer!

As of now, my objective is to create a point cloud comprising the original vertices and some additional ones to obtain the desired number. The triangles will come later.

I'll look into `blue_noise` right away.

Thanks!
